### PR TITLE
[NavBar] QA round 2

### PR DESCRIPTION
**Pull request by damassi**
More NavBar feedback:

1) New "tablet" mobile menu which has larger font, different spacing, etc: 

Screen Shot 2019-05-21 at 6 17 40 PM

2) It supports tap highlights: 

tap

3) Bigger open close icons on mobile: 

Screen Shot 2019-05-21 at 6 12 06 PM

4) Ensure top nav bar is 60px

### Misc other updates:
- Fixes weird missing icon / input text flash when SearchBar renders on the server
- Updated `react-tracking` to latest, which allows us to remove our custom `useTracking` hook code.
- Renamed `useMedia` hook to `useMatchMedia` and then created a new `useMedia` hook that's specifically tied to Palette's theme file and can be used like so: 

```tsx
import { useMedia } from "Utils/Hooks/useMedia"

const App = () => {
  const { xs, sm, md, lg, xl } = useMedia()

  return <div>Mobile view? {xs || sm}</div>
}

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit b6bbafc887cb731a9508111982bd6f8d18c503a1

https://deploy-preview-2447--artsy-reaction.netlify.com

File: src/Utils/Hooks/useMedia.ts [hunk 1]
```diff
@@ -1,12 +1,42 @@
...
+ *
+ * See: https://github.com/artsy/palette/blob/master/packages/palette/src/Theme.tsx#L84-L92
+ *
+ * @example
 
-import { useEffect, useState } from "react"
+    import { useMedia } from 'Utils/Hooks/useMedia'
+
+    const App = () => {
+      const { xs, sm, md, lg, xl } = useMedia()
```
Comment (damassi) [inline src/Utils/Hooks/useMedia.ts]: cc @pepopowitz

File: src/Components/Search/SearchBar.tsx [hunk 2]
```diff
@@ -441,14 +440,21 @@ export const SearchBarQueryRenderer: React.FC = () => {
...
-              placeholder={PLACEHOLDER_XS}
-            />
+            <>
+              <Box display={["block", "none"]}>
+                <SearchInputContainer placeholder={PLACEHOLDER_XS} />
+              </Box>
+              <Box display={["none", "block"]}>
+                <SearchInputContainer placeholder={PLACEHOLDER} />
+              </Box>
+            </>
```
Comment (damassi) [inline src/Components/Search/SearchBar.tsx]: @mzikherman / @pepopowitz - this fixes the weird SearchBar missing icon / text issue during the SSR render pass, when the searchbar first mounts: 

glass

File: package.json [hunk 3]
```diff
@@ -166,7 +166,6 @@
     "autosuggest-highlight": "^3.1.1",
     "cheerio": "^1.0.0-rc.2",
     "currency.js": "^1.2.1",
-    "deepmerge": "^3.2.0",
```
Comment (mzikherman) [inline package.json]: Intentional?

File: src/Components/NavBar/Menus/MobileNavMenu.tsx [hunk 4]
```diff
@@ -82,15 +82,39 @@ const MobileLink: React.FC<MobileLinkProps> = ({
   children,
   ...props
 }) => {
+  const { sm: isTablet } = useMedia()
+  const [isPressed, setPressed] = useState(false)
+  const bg = isTablet && isPressed ? "black5" : "white100"
+  const getEvents = () => {
```
Comment (mzikherman) [inline src/Components/NavBar/Menus/MobileNavMenu.tsx]: #minor but this probably doesn't need to be deferred? As in, doesn't need to be a function.

Comment (damassi) [reply src/Components/NavBar/Menus/MobileNavMenu.tsx]: It's for spreading in; it needs to return an empty object if a condition isn't met.

File: package.json (hunk 3 above)
Comment (damassi) [reply package.json]: Yup

**Comment by artsyit** [comment]
:rocket: PR was released in v16.12.7 :rocket:

**Pull request merged**